### PR TITLE
Test exception catching in test_missing_file

### PR DESCRIPTION
**Pull request by astrofrog**
I think this must be due to an update in pytest.

Fixes https://github.com/astropy/astropy/issues/9872

### CODE REVIEW

**Comment by astrofrog** [comment]
@pllim I've tagged you for review since you are a pytest exception/warning catching expert! :wink:

**Comment by astrofrog** [comment]
The Matplotlib dev failure is unrelated

File: astropy/visualization/wcsaxes/core.py [hunk 1]
```diff
@@ -464,7 +464,7 @@ def draw(self, renderer, inframe=False):
     # xlabel). While these are meant to be a single positional argument,
     # Matplotlib internally sometimes specifies e.g. set_xlabel(xlabel=...).
 
-    def set_xlabel(self, xlabel=None, labelpad=1, **kwargs):
+    def set_xlabel(self, xlabel=None, labelpad=1, loc=None, **kwargs):
```
Comment (pllim) [inline astropy/visualization/wcsaxes/core.py]: Why does `loc` break parallel build or is this for something else?

**Comment by pllim** [comment]
Looks uncontroversial enough but I'll wait for the CI to finish before approving. Thanks!

File: astropy/visualization/wcsaxes/core.py (hunk 1 above)
Comment (pllim) [reply astropy/visualization/wcsaxes/core.py]: OK, saw your second commit message. Nvm.

**Pull request merged**